pkg/restapi/errors/errors.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package errors
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+var logger = log.New("aries-framework/rest")
+
+// genericError is aries rest api error response
+// swagger:response genericError
+type genericError struct {
+	// in: body
+	Code Code `json:"code"`
+	// in: body
+	Message string `json:"message"`
+}
+
+// Code is the error code of aries rest api errors
+type Code int32
+
+const (
+	// UnknownStatus default error code for unknown errors
+	UnknownStatus Code = iota
+
+	// InvalidRequest is typically a validation error for invalid requests
+	InvalidRequest
+
+	// CreateInvitationError is for failures in create invitation endpoint
+	CreateInvitationError
+
+	// ReceiveInvitationError is for failures in receive invitation endpoint
+	ReceiveInvitationError
```

Comment:
These error codes are Protocol specific. Should we define these protocol specific error messages in common package (/pkg/restapi/errors) ?

Reply:
These are business errors, not at all protocol specific. There will be collision if we define them in separate packages.
I didn't export `genericError{}` indirectly forcing developers to use one of the `SendXXX()` functions to send errors.

Reply:
As discussed, I moved protocol errors to protocol specific implementations.
Introduced error Group and error code will be (Group + error code).
Example: DIDExchange will use `2000` series, Introduce will use `3000` series.